### DISCUSSION TITLE
Issue with "large" genomes?

### DISCUSSION THREAD

**Discussion by sgiorgetti** [GMOD/jbrowse-components#4119] [Q&A] [answered]
Hello guys.

Playing a bit with JBrowse2 and Apollo3, I felt bold and wanted to go for a not-so-small genome: GRCh38.p14

I could successfully load both the assembly and its related feature set, which I grabbed from Ensembl:
- [GRCh38.p14|https://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz]
- [Ensembl human 110|https://ftp.ensembl.org/pub/release-110/gff3/homo_sapiens/Homo_sapiens.GRCh38.110.chr_patch_hapl_scaff.gff3.gz]

I could see the features rendered in the UI, but when zooming in to see the sequence, I got a "Timeout error".
From the logs, we (@shashankbrgowda and I) can see the Apollo3 API is returning the expected results, but JBrowse2 is somehow not happy.
But we could not understand why.

In fairness, we are not using the latest version of JBrowse2 because we've had issues while creating an image for it.
Trying again to build the image with the latest one.

Meanwhile, may you advise? Is it anything you are aware of or stumbled upon?

Thanks,
S

**Comment by cmdcolin** [comment]
do you have a screenshot of the timeout error?

**Reply by sgiorgetti** [reply]
Sure thing! There you go. Thx
![image](https://github.com/GMOD/jbrowse-components/assets/78080390/fbc86f42-5c12-4738-8001-85fa175b8475)

**Reply by cmdcolin** [reply]
that is an odd error, I'm not exactly sure where it is coming from (it may be this is apollo specific). if you can open up the "browser devtools" (ctrl+shift+i) you might be able to see the "stack trace" for this error, and you can paste that here too

**Reply by sgiorgetti** [reply]
This is what I can see in the console. Hope it helps
```
BaseLinearDisplayModel.tsx:550 NonError: "timeout"
    at deserializeError (index.js:148:9)
    at t.value (client.js:123:44)
    at t.value (client.js:81:12)
(anonymous) @ BaseLinearDisplayModel.tsx:550
BaseLinearDisplayModel.tsx:550 NonError: "timeout"
    at deserializeError (index.js:148:9)
    at t.value (client.js:123:44)
    at t.value (client.js:81:12)
(anonymous) @ BaseLinearDisplayModel.tsx:550
serverSideRenderedBlock.ts:152 NonError: "timeout"
    at deserializeError (index.js:148:9)
    at t.value (client.js:123:44)
    at t.value (client.js:81:12)
setError @ serverSideRenderedBlock.ts:152
serverSideRenderedBlock.ts:152 NonError: "timeout"
    at deserializeError (index.js:148:9)
    at t.value (client.js:123:44)
    at t.value (client.js:81:12)
setError @ serverSideRenderedBlock.ts:152
```

**Comment by sgiorgetti** [comment, marked as answer]
Hello @cmdcolin - we managed to build the image for the latest JBrowse2 and this "fixed" the issue.

I am quoting "fixed" because we discovered an issue with Apollo3: it looks like it issues many API calls for the same region, hence hammering the DB, which takes a while to return the data.
Guess JBrowse2 - tired of waiting - may raise a timeout for this reason.

Anyway, not sure if it requires any intervention on JBrowse2: to me it would not, but it's your call.
The multiple API call issue in Apollo3 has been raised already.

I'll leave this discussion open, should you want to add something.
Happy to close it otherwise. Thanks!!!

**Reply by garrettjstevens** [reply]
The multiple API calls should be better after [e8f13a5](https://github.com/GMOD/Apollo3/commit/e8f13a57d7d7b8cbd8775841aeeb2fd9479cca04), which is in the current main branch of Apollo. If you're still running into the problem with the latest main branch, let me know.
